### PR TITLE
[4.0] Update the cookiecutter for JupyterLab 4 and Notebook 7

### PR DESCRIPTION
**Pull request by jtpio**
Fixes https://github.com/jupyterlab/extension-cookiecutter-ts/issues/231

The motivation is to be able to start creating extensions for JupyterLab 4 and Notebook 7 directly using this cookiecutter:

```
cookiecutter https://github.com/jupyterlab/extension-cookiecutter-ts --checkout 4.0
```

### CODE REVIEW

File: {{cookiecutter.python_name}}/binder/environment.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ channels:
 dependencies:
   # runtime dependencies
   - python >=3.8,<3.9.0a0
-  - jupyterlab >=3,<4.0.0a0
+  - jupyterlab >=4.0.0a29,<5
   # labextension build dependencies
   - nodejs >=14,<15
```
Comment (fcollonval) [inline {{cookiecutter.python_name}}/binder/environment.yml]: Should we upgrade nodejs (and maybe Python)?

```suggestion
  - nodejs >=16,<17
```

**Review by fcollonval — COMMENTED** [review]
I have one minor suggestion. Otherwise it looks good; thanks for this one @jtpio

File: {{cookiecutter.python_name}}/binder/environment.yml (hunk 1 above)
Comment (jtpio) [reply {{cookiecutter.python_name}}/binder/environment.yml]: Thanks, sounds good to me.

**Comment by jtpio** [comment]
Thanks @fcollonval for the review. I applied the suggestion and also update the Python version used on Binder.

**Pull request merged**